core/src/main/java/com/microsoft/alm/auth/oauth/AzureAuthority.java

```diff
@@ -249,6 +274,37 @@ public void call(final HttpURLConnection conn) {
         return tokens;
     }
 
+    /**
+     * Acquires an access token from the authority using a previously acquired refresh token.
+     *
+     * @param clientId     Identifier of the client requesting the token.
+     * @param resource     Identifier of the target resource that is the recipient of the requested token.
+     * @param refreshToken The {@link Token} of type {@link TokenType#Refresh}.
+     * @return If successful, a {@link TokenPair}; otherwise null.
+     */
+    public TokenPair acquireTokenByRefreshToken(final String clientId, final String resource,
+                                                final Token refreshToken) {
+        Debug.Assert(!StringHelper.isNullOrWhiteSpace(clientId), "The clientId parameter is null or empty");
+        Debug.Assert(!StringHelper.isNullOrWhiteSpace(resource), "The resource parameter is null or empty");
+        Debug.Assert(refreshToken!= null, "The refreshToken parameter is null");
+
+        logger.debug("AzureAuthority::acquireTokenByRefreshToken");
+
+        final URI tokenEndpoint = createTokenEndpointUri(authorityHostUrl);
+        final StringContent requestContent = createTokenRequestByRefreshToken(resource, clientId, refreshToken);
+
+        try {
+            final TokenPair tokenPair = doAcquireToken(tokenEndpoint, requestContent);
+            return tokenPair;
+        } catch (IOException e) {
+            // TODO: Follow the current standard which is return null when server calls failed.
```

Comment:
The original pattern I am following had a TODO in the block:
```
 // TODO: 449248: silently catching the exception here seems horribly wrong
 logger.debug("   token acquisition failed.");
 logger.debug("   IOException: {}", e);
```
Added this TODO to make sure we don't miss it when we cleanup.